unstable test TestPushDownSelectionForMPP

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
 --- FAIL: TestPushDownSelectionForMPP (10.71s)
    result.go:49: 
        	Error Trace:	/go/tidb/planner/core/result.go:49
        	            				/go/tidb/planner/core/integration_test.go:4323
        	Error:      	Not equal: 
        	            	expected: "[TableReader 6400.00 root  data:ExchangeSender]\n[└─ExchangeSender 6400.00 mpp[tiflash]  ExchangeType: PassThrough]\n[  └─Selection 6400.00 mpp[tiflash]  gt(test.t.id, Column#5)]\n[    └─Projection 8000.00 mpp[tiflash]  Column#5, test.t.id]\n[      └─HashAgg 8000.00 mpp[tiflash]  group by:test.t.id, funcs:count(1)->Column#5, funcs:firstrow(test.t.id)->test.t.id]\n[        └─ExchangeReceiver 10000.00 mpp[tiflash]  ]\n[          └─ExchangeSender 10000.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.t.id, collate: binary]]\n[            └─TableFullScan 10000.00 mpp[tiflash] table:t keep order:false, stats:pseudo]\n"
        	            	actual  : "[Selection 6400.00 root  gt(test.t.id, Column#5)]\n[└─HashAgg 8000.00 root  group by:test.t.id, funcs:count(1)->Column#5, funcs:firstrow(test.t.id)->test.t.id]\n[  └─TableReader 10000.00 root  data:TableFullScan]\n[    └─TableFullScan 10000.00 cop[tikv] table:t keep order:false, stats:pseudo]\n"
        	            	
1ms]
[2022/08/03 21:07:55.943 +08:00] [INFO] [domain.go:154] ["diff load InfoSchema success"] [currentSchemaVersion=319] [neededSchemaVersion=336] ["start time"=1.45397ms] [phyTblIDs="[]"] [actionTypes="[]"]
[2022/08/03 21:07:55.943 +08:00] [WARN] [domain.go:464] ["loading schema takes a long time"] ["take time"=2.333842ms]
[2022/08/03 21:07:55.946 +08:00] [INFO] [domain.go:154] ["diff load InfoSchema success"] [currentSchemaVersion=319] [neededSchemaVersion=336] ["start time"=1.456713ms] [phyTblIDs="[]"] [actionTypes="[]"]
[2022/08/03 21:07:55.946 +08:00] [WARN] [domain.go:464] ["loading schema takes a long time"] ["take time"=3.00376ms]
[2022/08/03 21:07:55.949 +08:00] [INFO] [domain.go:154] ["diff load InfoSchema success"] [currentSchemaVersion=319] [neededSchemaVersion=336] ["start time"=1.613267ms] [phyTblIDs="[]"] [actionTypes="[]"]
[2022/08/03 21:07:55.949 +08:00] [WARN] [domain.go:464] ["loading schema takes a long time"] ["take time"=2.656009ms]
[2022/08/03 21:07:55.952 +08:00] [INFO] [domain.go:154] ["diff load InfoSchema success"] [currentSchemaVersion=319] [neededSchemaVersion=336] ["start time"=1.670245ms] [phyTblIDs="[]"] [actionTypes="[]"]
[2022/08/03 2
        	            	+[  └─TableReader 10000.00 root  data:TableFullScan]
        	            	+[    └─TableFullScan 10000.00 cop[tikv] table:t keep order:false, stats:pseudo]
        	            	 
        	Test:       	TestPushDownSelectionForMPP
        	Messages:   	sql:desc format = 'brief' select /*+ hash_agg()*/ count(*) c, id from t group by id having id >c, args:[]
FAIL
```

https://prow.pingcap.net/view/gs/pingcapprow/logs/tidb_race/1554807036774977536
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/planner/core/integration_test.go
+++ b/planner/core/integration_test.go
@@ -4306,6 +4306,7 @@ func TestPushDownSelectionForMPP(t *testing.T) {
 	}
 
 	tk.MustExec("set @@tidb_allow_mpp=1; set @@tidb_enforce_mpp=1;")
+	tk.MustExec("set @@tidb_isolation_read_engines='tiflash,tidb';")
 
 	var input []string
 	var output []struct {

